framework/widgets/ActiveField.php

```diff
@@ -775,19 +775,19 @@ public function widget($class, $config = [])
         $config['attribute'] = $this->attribute;
         $config['view'] = $this->form->getView();
         if (is_subclass_of($class, 'yii\widgets\InputWidget')) {
-            foreach ($this->inputOptions as $key => $value) {
-                if (!isset($config['options'][$key])) {
-                    $config['options'][$key] = $value;
-                }
-            }
+            $classOptions = (new \ReflectionClass($class))->getDefaultProperties();
+            $classOptions = is_null($classOptions['options']) ? [] : $classOptions['options'];
+
+            $definitions = \Yii::$container->getDefinitions();
```

Comment:
Direct access to container in the widget class is a bit hacky. Is it possible to avoid it?

Reply:
Agree. Any suggestions on how to avoid it?

Reply:
No, no immediate suggestions.